### PR TITLE
Add Zero Clause BSD license

### LINKED ISSUE #642: Add 0BSD License
https://github.com/spdx/license-list-XML/pull/768

screen shot 2019-02-11 at 6 16 24 pm

This seems to be missing from your list, which I was under the impression was pulled from SPDX license XML (it's already there).

Ref: #464

### PR DESCRIPTION
**Pull request by ghost**
close #642

### CODE REVIEW

**Comment by waldyrious** [comment]
Shall the copyright line be included in the license text (with a generic name like [this](https://github.com/github/choosealicense.com/blob/44260e9c6/_licenses/mit.txt#L33)), per @landley's [comment](https://github.com/spdx/license-list/issues/23#issuecomment-463478175) in https://github.com/spdx/license-list/issues/23?

**Comment by mlinksva** [comment]
Yes there should be a copyright line, also to match the SPDX text as closely as possible https://github.com/github/choosealicense.com/issues/636

@jhabdas there are two test failures:

1. the name doesn't match what is in SPDX https://travis-ci.org/github/choosealicense.com/builds/493146734#L1301-L1305
2. one of the examples wouldn't be detected https://travis-ci.org/github/choosealicense.com/builds/493146734#L1310-L1314 ... for that one, the problem is almost certainly the first line. Find an example that doesn't have such additions to the text (licensee can't tell whether such additions are nefarious or not, so conservatively does not give a positive identification of a license with added text)

**Comment by landley** [comment]
That's my project, I can yank the intro line if it would help?

**Comment by landley** [comment]
Done. Try now?

**Comment by mlinksva** [comment]
@landley thanks, works now!

**Comment by jhabdas** [comment]
Thanks. I'm on +8:00 and had to run to immigration today. Sounds like we've got progress. But I'm not sure what's left.

> 1. the name doesn't match what is in SPDX https://travis-ci.org/github/choosealicense.com/builds/493146734#L1301-L1305

Opened a pull a [few days ago](https://github.com/spdx/license-list-XML/pull/768) to have the title adjusted at the SPDX end. Once that blocker is cleared CI should pass.

**Comment by jhabdas** [comment]
> 1. the name doesn't match what is in SPDX https://travis-ci.org/github/choosealicense.com/builds/493146734#L1301-L1305

Opened a pull a [few days ago](https://github.com/spdx/license-list-XML/pull/768) to have the title adjusted at the SPDX end. Once that blocker is cleared CI should pass. Not sure who to ping but I do know I had issues merging because someone updated the same LOC after my pull was submitted but hasn't commented on my PR.

**Comment by jhabdas** [comment]
> 1. the name doesn't match what is in SPDX https://travis-ci.org/github/choosealicense.com/builds/493146734#L1301-L1305

Opened a pull a [few days ago](https://github.com/spdx/license-list-XML/pull/768) to have the title adjusted at the SPDX end. Sorry for the mess in the pull I was working with an older copy of the source.

**Comment by jhabdas** [comment]
> 1. the name doesn't match what is in SPDX https://travis-ci.org/github/choosealicense.com/builds/493146734#L1301-L1305

Opened a pull a [few days ago](https://github.com/spdx/license-list-XML/pull/768) to have the title adjusted at the SPDX end. Once that's merged CI should start passing if I'm reading Travis properly.

**Comment by ghost** [comment]
> 1. the name doesn't match what is in SPDX https://travis-ci.org/github/choosealicense.com/builds/493146734#L1301-L1305

Opened a pull a [few days ago](https://github.com/spdx/license-list-XML/pull/768) to have the title adjusted at the SPDX end. Once that's merged CI should start passing if I'm reading Travis properly. I'll get the copyright line added here in a jiff. **Edit:** Done.

@waldyrious I remember you mentioning adding the license title to the top of the license. ~~I left it out as it's not part of [the license text](https://spdx.org/licenses/0BSD.html). I maintain no opinion as to its inclusion in this pull request but personally don't plan to use it.~~ I contradicted myself there. I've added the title of the license.

**Comment by jhabdas** [comment]
@waldyrious I remember you mentioning adding the license title to the top of the license. I left it out as it's not part of [the license text](https://spdx.org/licenses/0BSD.html). I maintain no opinion as to its inclusion in this pull request but personally don't plan to use it.

**Comment by landley** [comment]
Please just let SPDX stay how it is and change the title here if you need to match it. I don't see a huge difference between "Zero Clause BSD" and "BSD Zero Clause" (with or without hyphens), and would rather avoid the churn. It's been stable in SPDX for several years now, the stability itself means something.

**Comment by landley** [comment]
If anything in SPDX needed to be fixed, it's that even though I submitted it to their mailing list with "Your Name" in the license's copyright line, the copy at https://spdx.org/licenses/0BSD.html is cut and pasted from my toybox project's license page with _my_ name in the copyright line. (I hadn't considered it worth the churn to fix, but if somebody else wants to...)

(Alas SPDX deleted its mailing list archives, and google still pulls up https://lists.spdx.org/pipermail/spdx-legal/2015-June which is 404, but I still have the copy in my outbox. I used the same copyright line the "license template" link from https://www.openbsd.org/policy.html has, since the Open BSD suggested template license was the upstream source I modified to create 0BSD in the first place. I went back to the source for the standards body submission...)

**Comment by mlinksva** [comment]
@landley the SPDX list archives moved but sadly the old links were not cool enough to redirect, I guess. Anyway: https://lists.spdx.org/g/spdx/topic/22079600

@jhabdas thanks for pushing this forward.

I made as consistent as possible with what's in SPDX at this point and will merge soon. Tests are only failing at this moment due to unrelated external links being flakey.

**Comment by ghost** [comment]
Thanks to all. I took the liberty of adding this to Wikipedia. @landley please feel free to make any adjustments you see fit. https://en.wikipedia.org/wiki/BSD_licenses

**Comment by landley** [comment]
@mlinksva That was my first post about it, asking if there was a form to fill out. They pointed me at the correct procedure document and I did a second post in their official format, and that's where I used the "your name" copyright. Alas I can't navigate their new archive to find it.

@jabdas Yes, thanks for pushing it forward.

Re: wikipedia: I can't edit wikipedia because they don't want anyone with first hand knowledge collapsing their quantum state. It would jeopardize their status as the world's largest accumulation of anecdotal evidence. (The closest I come is ading things to the "talk" pages of articles.)

But please stop deadnaming the license. It was called Zero clause BSD _first_ (by multiple years on the toybox website), was approved and published by SPDX before it was ever even _submitted_ to OSI, and even OSI has now switched to more or less conform to SPDX. (Modulo word order and a hyphen.)

While creating the new license I asked Marshall McKusick (when he was keynote speaker at Ohio LinuxFest in 2013) if he was ok calling a derivative of OpenBSD's suggested Template License "Zero Clause BSD" and he said yes, and while convincing OSI to undo their mistake I asked him again if he remembered and got an explicit statement of support from him (I attached the full email to http://lists.landley.net/pipermail/toybox-landley.net/2018-October/009780.html although dreamhost's web server thinks "blah.eml" is a format to download rather than ascii text to display, which I can't fix because dreamhost, but feel free to copy it somewhere more convenient if you like).

Seriously, the OSI misnaming was a mistake and it's over now. Please don't memorialize it, it never served any purpose except to confuse people and even OSI eventually admitted that and fixed it.

**Comment by landley** [comment]
Um, it says "merged". How do I tell when it's identified 0bsd in https://github.com/landley/toybox (is there something to look for on that page? Do I need to modify the file for it to reevaluate it?) and when (or if) does https://help.github.com/articles/licensing-a-repository/ update? (I dunno the procedure here...)

Thank you very much.

**Comment by mlinksva** [comment]
@landley it'll take awhile, see https://github.com/github/choosealicense.com/issues/642#issuecomment-463318223

I'll add a note here when github.com knows about 0BSD, but if you wish to test yourself with a cron job o something, it'll be when https://api.github.com/licenses/0bsd doesn't 404.

**Comment by jhabdas** [comment]
> But please stop deadnaming the license.

@landley Ideally there would be one name. Practically it's the telephone game. Just tell me what you want Wikipedia to saw WRT the name and consider it done.

**Comment by ghost** [comment]
> But please stop deadnaming the license.

@landley Ideally there would be one name. Practically it's the telephone game. Just tell me what you want Wikipedia to say WRT the name and consider it done.

**Comment by landley** [comment]
Just the new name please. The controversy is resolved. If you'd like me to post the controversy to the talk section, I can, but these days it only serves to confuse matters.

Kirk McKusick (the maintainer Bill Joy handed BSD off to at Berkeley) has endorsed the new name (twice). The guy who submitted it to OSI under the other name has acknowledged the new name (also twice). OSI took down the old page and put up a new page with the correct name. SPDX's approval came significantly before OSI's submission, and if you're going by dinosaur naming rules I was publicly using the name in toybox something like a year before I sent it to SPDX (and spoke to McKusick about it at Ohio Linux Symposium 2013, before that, showing I already _had_ the name by then and was clearing it).

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, significantly more accessible and I'm confident it will be replacing the MIT in years to come.

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, significantly more accessible and I'm confident it will be gaining greater adoption quickly.

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, more accessible than most other licenses and I'm confident it will be gaining greater adoption quickly.

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, more accessible than most other licenses and I'm confident it will be gaining even greater adoption quickly.

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, more accessible than most other licenses I've seen and I'm confident it will continue greater adoption given it fixes what WTFPL got wrong.

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, more accessible than most other licenses I've seen and I'm confident it will continue greater adoption for years to come.

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, more accessible than most other licenses I've seen and I'm confident it will gain wide adoption in years to come.

**Comment by jhabdas** [comment]
At any rate, thanks. This license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, more accessible than most other licenses I've seen and I'm confident it will gain even wider adoption in years to come.

**Comment by ghost** [comment]
At any rate, thanks. Your license lexicographically sorts to the top and bottom of lists now depending on if the list is sorted by SPDX identifier or name. And I'm confident that exposure will help prevent that nagging doubt FOSS devs feel when they start pulling in bits and pieces of otherwise _free_ code into their projects and have to [question authors](https://github.com/danklammer/bytesize-icons/issues/7) about how to use it. 0BSD is ultra clear, more accessible than most other licenses I've seen and I'm confident it will gain even wider adoption in years to come.

**Review by presiyan6340 — COMMENTED** [review]
pose

File: about.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ See our [appendix](/appendix/) for a table of all of the licenses cataloged in t
 {: .bullets}
 
 * Open Source Initiative's FAQ on [Which Open Source license should I choose to release my software under?](https://opensource.org/faq#which-license)
-* Free Software Foundation's [advice on how to choose a license](https://www.gnu.org/licenses/license-recommendations.en.html)
+* Free Software Foundation's [advice on how to choose a license](https://www.gnu.org/licenses/license-recommendations.html)
```
Comment (presiyan6340) [inline about.md:22]: (n)

File: Rakefile [hunk 2]
```diff
@@ -17,6 +17,7 @@ task :test do
                               enforce_https: true,
                               validation: { ignore_script_embeds: true },
                               url_swap: { %r{https://choosealicense.com} => '' },
+                              url_ignore: [%r{https://github.com/github/choosealicense.com/edit/gh-pages/_licenses/}],
                               hydra: { max_concurrency: 10 },
                               check_img_http: true).run
 end
```
Comment (presiyan6340) [inline Rakefile:23]: (n2)

Comment (presiyan6340) [inline Rakefile:23]: (n-2))

**Pull request merged**